torch_ttnn/passes/lowering/add_data_move_pass.py

```diff
@@ -315,252 +287,240 @@ def is_target_a_user_of_curr_node(curr_node, target):
+                aligning_nodes.append(g.call_function(ttnn.from_device, (spec.input_node,)))
+            if need_to_layout:
+                aligning_nodes.append(
+                    g.call_function(
+                        ttnn.to_layout, (aligning_nodes[-1] if aligning_nodes else spec.input_node, spec.layout())
+                    )
+                )
+            if need_to_device:
+                aligning_nodes.append(
+                    g.call_function(
+                        ttnn.to_device,
+                        (aligning_nodes[-1] if aligning_nodes else spec.input_node,),
+                        {"device": spec.device()},
+                    )
+                )
+        return aligning_nodes[-1]
+
+    def _connect_aligned_node(self, node, aligned_node, input_site, input_site_type):
+        if input_site_type == "args":
+            input_idx = input_site
+            node.update_arg(input_idx, aligned_node)
+        elif input_site_type == "kwargs":
+            key = input_site
+            node.update_kwarg(key, aligned_node)
+        elif input_site_type == "args_tuple":
+            input_idx = input_site[0]
+            tuple_idx = input_site[1]
+            old_arg = node.args[input_idx]
+            new_arg = list(old_arg)
+            new_arg[tuple_idx] = aligned_node
+            node.update_arg(input_idx, tuple(new_arg))
+        elif input_site_type == "kwargs_tuple":
+            key = input_site[0]
+            tuple_idx = input_site[1]
+            old_arg = node.kwargs[key]
+            new_arg = list(old_arg)
+            new_arg[tuple_idx] = aligned_node
+            node.update_kwarg(key, tuple(new_arg))
+
+    def _connect_aligned_node_layer_norm(self, node, input_node, aligned_node, input_site, input_site_type):
+        # Workaround to output the same layer_norm output
+        # Before: layer_norm = aten.layer_norm
+        #          getitem = getitem(layer_norm, 0)
+        #          return ((getitem,),)
+        # After: layer_norm = ttnn.layer_norm
+        #        return (layer_norm,)
+        # Need to match the tuple in the original return statement
+        old_args = node.args[0]
         if isinstance(old_args, tuple):
             new_args = list(old_args)
             for idx, old_arg in enumerate(old_args):
-                if old_arg == src_node:
-                    new_args[idx] = new_nodes[-1]
-            dst_node.update_arg(0, tuple(new_args))
+                if old_arg == input_node:
+                    new_args[idx] = aligned_node
+            node.update_arg(0, tuple(new_args))
         else:
-            dst_node.update_arg(dst_idx, new_nodes[-1])
-        return new_nodes[-1]
-    else:
-        return None
+            self._connect_aligned_node(node, aligned_node, input_site, input_site_type)
+
+    def align(self, node, input_node, input_site, input_site_type):
+        assert input_site_type in ["args", "kwargs", "args_tuple", "kwargs_tuple"]
```

Comment:
Maybe we can define enum for these values https://docs.python.org/3/library/enum.html. So it is harder to have typo with them in places above

Reply:
yes, I've use the Enum to ensure